File: payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/events/ChargeCreatedEvent.java
package com.hedvig.paymentservice.domain.payments.events;

import lombok.Value;
import org.axonframework.serialization.Revision;

import javax.money.MonetaryAmount;
import java.time.Instant;
import java.util.UUID;

@Value
@Revision("1.0")
public class ChargeCreatedEvent {
  String memberId;
  UUID transactionId;
  MonetaryAmount amount;
  Instant timestamp;
  String accountId; //providerId
  //provider ENUM
  String email;
  String createdBy;
}

File: payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/events/ChargeCreatedEvent.kt
package com.hedvig.paymentservice.domain.payments.events

import com.hedvig.paymentservice.domain.payments.enums.PayinProvider
import java.time.Instant
import java.util.UUID
import javax.money.MonetaryAmount
import org.axonframework.serialization.Revision

@Revision("2.0")
data class ChargeCreatedEvent(
    val memberId: String,
    val transactionId: UUID,
    val amount: MonetaryAmount,
    val timestamp: Instant,
    val providerId: String,
    val provider: PayinProvider,
    val email: String,
    var createdBy: String
)
